src/background/injector-controller.ts

```diff
@@ -61,6 +61,7 @@ export class InjectorController {
                 });
             }, InjectorController.injectionStartedWaitTime);
 
+            // tslint:disable-next-line:no-floating-promises - grandfathered
```

Comment:
This is nice to see, I didn't realize the use of Q was masking valid lint failures. Makes me feel happy that you've picked up this work :)

Reply:
I'd actually wonder if this is something we actually want to properly handle as this handled promise is not on any of the extension's entry point. I decided to grandfathered the issue only because the main concern of this PR was moving away from Q.